app/services/authentication_service.rb

```diff
@@ -0,0 +1,19 @@
+# We do not want to store the actual slack id in order to
+# keep users anonymous. Instead we use SHA256 to create a hashed
+# version of the id and store that in the DB. SHA256
+# is a one way, deterministic hashing algorithm. It is impossible to
+# go from hashed_slack_id back to the original id. It is only possible
+# to go from slack_id to hashed_slack_id.
```

Comment:
These comments and the new section in the readme are super clear! I trust that it'll be easy for anyone jumping into this codebase to understand why it's set up this way.